### PR TITLE
DDF-3433 updated definition of DDF_HOME to work with whitespace

### PR DESCRIPTION
**Pull request by oconnormi**
#### What does this PR do?

Moves the definition of DDF_HOME out of setenv and into etc/system.properties. This solves the issue of running DDF under a directory with whitespace in the name.

#### Who is reviewing it? 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged)
@emmberk @AzGoalie @peterhuffer 

#### Choose 2 committers to review/merge the PR. 

@clockard
@coyotesqrl

#### How should this be tested? (List steps with links to updated documentation)

Build and start DDF on both linux and windows with whitespace in the directory name.
Ensure that DDF starts up and installs successfully.

#### Any background context you want to provide?

Paths with spaces were not being expanded properly due to the combination of how they were being discovered/saved in the `setenv` script and then referenced in the `karaf` script. This was causing the first term after any whitespace in the path to be interpreted as the classname.

Since `KARAF_HOME` is already discovered and passed in correctly by the `karaf` script the java property for `ddf.home` could be set inside of `system.properties` which would work on both windows and *nix

#### What are the relevant tickets?
[DDF-3433](https://codice.atlassian.net/browse/DDF-3433)

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by oconnormi** [comment]
build now

**Comment by oconnormi** [comment]
@paouelle I updated the additional files, I'm going to start testing them on both linux and windows now to make sure everything works correctly

**Comment by oconnormi** [comment]
Build Now

**Comment by AzGoalie** [comment]
verified on windows 👍

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf [hunk 1]
```diff
@@ -51,15 +49,15 @@ wrapper.java.additional.26 = -XX:OnError=%KARAF_HOME%/bin/ddf_on_error.sh wrappe
...
-# wrapper.java.additional.30 = -Djava.security.manager=net.sourceforge.prograde.sm.PolicyFileGeneratorJSM
-# wrapper.java.additional.31 = -Dprograde.generated.policy=%KARAF_HOME%/generated.policy
-# wrapper.java.additional.32 = -Dprograde.use.own.policy=true
+# wrapper.java.additional.28 = -Djava.security.manager=net.sourceforge.prograde.sm.PolicyFileGeneratorJSM
+# wrapper.java.additional.29 = -Dprograde.generated.policy=%KARAF_HOME%/generated.policy
+# wrapper.java.additional.30 = -Dprograde.use.own.policy=true
 #
 # Production Security Permissions
-# wrapper.java.additional.30 = -Djava.security.manager=net.sourceforge.prograde.sm.ProGradeJSM
+# wrapper.java.additional.31 = -Djava.security.manager=net.sourceforge.prograde.sm.ProGradeJSM
```
Comment (paouelle) [inline distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: This should be 28

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf [hunk 2]
```diff
@@ -51,15 +49,15 @@ wrapper.java.additional.26 = -XX:OnError=%KARAF_HOME%/bin/ddf_on_error.sh wrappe
...
 
 # Uncomment to enable cxf logging interceptors
-# wrapper.java.additional.31 = -Dcom.sun.xml.ws.transport.http.HttpAdapter.dump=true
+# wrapper.java.additional.32 = -Dcom.sun.xml.ws.transport.http.HttpAdapter.dump=true
```
Comment (paouelle) [inline distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: this should be 29

File: distribution/ddf-common/src/main/resources/bin/setenv.bat [hunk 3]
```diff
@@ -100,5 +99,6 @@ rem Defines the special on-error Java options
 rem set JAVA_ERROR_OPTS=-XX:OnOutOfMemoryError=%DDF_ON_ERROR%%%p -XX:OnError=%DDF_ON_ERROR%%%p
 set JAVA_ERROR_OPTS=-XX:OnOutOfMemoryError=%DDF_ON_ERROR% -XX:OnError=%DDF_ON_ERROR%
 
-set KARAF_OPTS=-Dfile.encoding=UTF8 -Dddf.home=%DDF_HOME% -Dddf.home.policy=%DDF_HOME_POLICY% 
-set JAVA_OPTS=-Xms2g -Xmx4g -Dderby.system.home="%DDF_HOME%\data\derby" -Dderby.storage.fileSyncTransactionLog=true -XX:+DisableAttachMechanism %JAVA_ERROR_OPTS%
+set KARAF_OPTS=-Dfile.encoding=UTF8
+
+set JAVA_OPTS=-Xms2g -Xmx4g -Dderby.storage.fileSyncTransactionLog=true -Dfile.encoding=UTF8 -XX:+DisableAttachMechanism
```
Comment (paouelle) [inline distribution/ddf-common/src/main/resources/bin/setenv.bat]: missing the %JAVA_ERROR_OPTS%

**Comment by AzGoalie** [comment]
verified on windows 👍

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf [hunk 4]
```diff
@@ -12,36 +12,34 @@ wrapper.commandfile = %KARAF_HOME%/bin/restart.jvm
...
+wrapper.java.additional.21 = -Dorg.tanukisoftware.wrapper.WrapperManager.mbean=TRUE
 
 # Prefer /dev/urandom over /dev/random on Linux systems.
-wrapper.java.additional.24 = -Djava.security.egd=file:/dev/./urandom
+wrapper.java.additional.22 = -Djava.security.egd=file:/dev/./urandom
 
 # Error handling scripts
-wrapper.java.additional.25 = -XX:OnOutOfMemoryError=%KARAF_HOME%/bin/ddf_on_error.sh wrapper %p
-wrapper.java.additional.26 = -XX:OnError=%KARAF_HOME%/bin/ddf_on_error.sh wrapper %p
+wrapper.java.additional.23 = -XX:OnOutOfMemoryError=%KARAF_HOME%/bin/ddf_on_error.sh wrapper %p
```
Comment (paouelle) [inline distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: ❓ Will this be an issue with spaces?

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf [hunk 5]
```diff
@@ -12,36 +12,34 @@ wrapper.commandfile = %KARAF_HOME%/bin/restart.jvm
...
+wrapper.java.additional.24 = -XX:OnError=%KARAF_HOME%/bin/ddf_on_error.sh wrapper %p
```
Comment (paouelle) [inline distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: ❓ Will this be an issue with spaces?

File: distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf [hunk 6]
```diff
@@ -12,33 +12,31 @@ wrapper.commandfile = %KARAF_HOME%/bin/restart.jvm
...
+wrapper.java.additional.20 = -Djava.awt.headless=true
 
 # Enables MBean support for the service wrapper; used to trigger restart
-wrapper.java.additional.23 = -Dorg.tanukisoftware.wrapper.WrapperManager.mbean=TRUE
+wrapper.java.additional.21 = -Dorg.tanukisoftware.wrapper.WrapperManager.mbean=TRUE
 
 # Error handling scripts
-wrapper.java.additional.24 = "-XX:OnOutOfMemoryError=%KARAF_HOME%\bin\ddf_on_error.bat wrapper %p"
-wrapper.java.additional.25 = "-XX:OnError=%KARAF_HOME%\bin\ddf_on_error.bat wrapper %p"
+wrapper.java.additional.22 = "-XX:OnOutOfMemoryError=%KARAF_HOME%\bin\ddf_on_error.bat wrapper %p"
```
Comment (paouelle) [inline distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf]: ❓ Will this be an issue with spaces?

File: distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf [hunk 7]
```diff
@@ -12,33 +12,31 @@ wrapper.commandfile = %KARAF_HOME%/bin/restart.jvm
...
+wrapper.java.additional.23 = "-XX:OnError=%KARAF_HOME%\bin\ddf_on_error.bat wrapper %p"
```
Comment (paouelle) [inline distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf]: ❓ Will this be an issue with spaces?

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf [hunk 8]
```diff
@@ -12,36 +12,34 @@ wrapper.commandfile = %KARAF_HOME%/bin/restart.jvm
...
-wrapper.java.additional.18 = -Djava.util.logging.config.file=%KARAF_HOME%/etc/java.util.logging.properties
-wrapper.java.additional.19 = -XX:+UnlockDiagnosticVMOptions
-wrapper.java.additional.20 = -XX:+DisableAttachMechanism
-wrapper.java.additional.21 = -XX:+UnsyncloadClass
-wrapper.java.additional.22 = -Djava.awt.headless=true
+wrapper.java.additional.11 = -server
+wrapper.java.additional.12 = -Dfile.encoding=UTF8
+wrapper.java.additional.13 = -Dkaraf.instances=%KARAF_HOME%/instances
+wrapper.java.additional.14 = -Dkaraf.restart.jvm.supported=true
+wrapper.java.additional.15 = -Djava.io.tmpdir=%KARAF_HOME%/data/tmp
```
Comment (paouelle) [inline distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: ❓ Will this be an issue with spaces?

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf [hunk 9]
```diff
@@ -12,36 +12,34 @@ wrapper.commandfile = %KARAF_HOME%/bin/restart.jvm
...
+wrapper.java.additional.16 = -Djava.util.logging.config.file=%KARAF_HOME%/etc/java.util.logging.properties
```
Comment (paouelle) [inline distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: ❓ Will this be an issue with spaces?

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf [hunk 10]
```diff
@@ -12,36 +12,34 @@ wrapper.commandfile = %KARAF_HOME%/bin/restart.jvm
...
-wrapper.java.additional.16 = -Dkaraf.restart.jvm.supported=true
-wrapper.java.additional.17 = -Djava.io.tmpdir=%KARAF_HOME%/data/tmp
-wrapper.java.additional.18 = -Djava.util.logging.config.file=%KARAF_HOME%/etc/java.util.logging.properties
-wrapper.java.additional.19 = -XX:+UnlockDiagnosticVMOptions
-wrapper.java.additional.20 = -XX:+DisableAttachMechanism
-wrapper.java.additional.21 = -XX:+UnsyncloadClass
-wrapper.java.additional.22 = -Djava.awt.headless=true
+wrapper.java.additional.11 = -server
+wrapper.java.additional.12 = -Dfile.encoding=UTF8
+wrapper.java.additional.13 = -Dkaraf.instances=%KARAF_HOME%/instances
```
Comment (paouelle) [inline distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: ❓ Will this be an issue with spaces?

File: distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf [hunk 11]
```diff
@@ -12,33 +12,31 @@ wrapper.commandfile = %KARAF_HOME%/bin/restart.jvm
...
-wrapper.java.additional.16 = -Dkaraf.restart.jvm.supported=true
-wrapper.java.additional.17 = -Djava.io.tmpdir=%KARAF_HOME%/data/tmp
-wrapper.java.additional.18 = -Djava.util.logging.config.file=%KARAF_HOME%/etc/java.util.logging.properties
-wrapper.java.additional.19 = -XX:+UnlockDiagnosticVMOptions
-wrapper.java.additional.20 = -XX:+DisableAttachMechanism
-wrapper.java.additional.21 = -XX:+UnsyncloadClass
-wrapper.java.additional.22 = -Djava.awt.headless=true
+wrapper.java.additional.11 = -server
+wrapper.java.additional.12 = -Dfile.encoding=UTF8
+wrapper.java.additional.13 = -Dkaraf.instances=%KARAF_HOME%/instances
```
Comment (paouelle) [inline distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf]: ❓ Will this be an issue with spaces?

File: distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf [hunk 12]
```diff
@@ -12,33 +12,31 @@ wrapper.commandfile = %KARAF_HOME%/bin/restart.jvm
...
+wrapper.java.additional.14 = -Dkaraf.restart.jvm.supported=true
+wrapper.java.additional.15 = -Djava.io.tmpdir=%KARAF_HOME%/data/tmp
```
Comment (paouelle) [inline distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf]: ❓ Will this be an issue with spaces?

File: distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf [hunk 13]
```diff
@@ -12,33 +12,31 @@ wrapper.commandfile = %KARAF_HOME%/bin/restart.jvm
...
+wrapper.java.additional.16 = -Djava.util.logging.config.file=%KARAF_HOME%/etc/java.util.logging.properties
```
Comment (paouelle) [inline distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf]: ❓ Will this be an issue with spaces?

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf (hunk 10 above)
Comment (oconnormi) [reply distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: Nope, service wrapper can handle paths with spaces, and actually requires that they not be quoted

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf (hunk 8 above)
Comment (oconnormi) [reply distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: See above

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf (hunk 9 above)
Comment (oconnormi) [reply distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: See above

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf (hunk 4 above)
Comment (oconnormi) [reply distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: See above

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf (hunk 5 above)
Comment (oconnormi) [reply distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: See above

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf (hunk 1 above)
Comment (oconnormi) [reply distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: I'm replacing all the commented out line numbers with `.<Number>` and a comment in the file to replace that value with the next available number

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf (hunk 2 above)
Comment (oconnormi) [reply distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: See above

File: distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf (hunk 12 above)
Comment (oconnormi) [reply distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf]: See above

File: distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf (hunk 13 above)
Comment (oconnormi) [reply distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf]: See above

File: distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf (hunk 6 above)
Comment (oconnormi) [reply distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf]: See above

File: distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf (hunk 7 above)
Comment (oconnormi) [reply distribution/ddf-common/src/main/resources/bin/setenv-windows-wrapper.conf]: See above

File: distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf (hunk 10 above)
Comment (paouelle) [reply distribution/ddf-common/src/main/resources/bin/setenv-wrapper.conf]: Thanks for checking

**Comment by oconnormi** [comment]
Build Now

**Comment by oconnormi** [comment]
Build Now

**Comment by oconnormi** [comment]
Build Now

**Comment by oconnormi** [comment]
@coyotesqrl I think should actually work with security manager once its enabled. At least judging by how karaf enables it in their project: https://github.com/apache/karaf/blob/master/assemblies/features/base/src/main/resources/resources/etc/system.properties#L135-L137

I'm not very familiar with how our security manager works, but if it is at all like the one included in karaf and just requires that `java.security.policy` be set to point at the policy file then it should be fine. Does that sound right?

**Comment by coyotesqrl** [comment]
@oconnormi The policy file needs to be addressed prior to startup of the JVM. That could be managed without setting `ddf.home` explicitly; however, there are substitution params inside the policy file itself that need to be set prior to startup as well. They need to be in the environment early.

**Comment by oconnormi** [comment]
@coyotesqrl I just updated this to move the security manager properties into `system.properties`. I have tested this with security manager enabled by uncommenting the properties prior to startup. I can confirm that the default policy is working, if i remove any grants from the `${ddf.home}/security/default.policy` file I see failures very quickly during boot. Are there any additional things I should check to confirm that this will work?

**Comment by oconnormi** [comment]
Build Now

**Comment by coyotesqrl** [comment]
@oconnormi That should be sufficient for now. There will be significant changes to startup when the time comes to integrate the security manager; any issues can be addressed then.

**Comment by oconnormi** [comment]
Build Now

**Comment by oconnormi** [comment]
hmm, something went wrong with the mirror. trying again, Build Now

**Pull request merged**